### ISSUE TITLE
JRE link to oracle site broken

### ISSUE THREAD

**Issue by IanWardell** [gethvi/iDRAC6VirtualConsoleLauncher#1]
Issue: Broken java jre link 
Correction: new link  
https://download.oracle.com/otn/java/jdk/7u80-b15/jre-7u80-linux-x64.tar.gz

Notes:
Great python script for this, I found it from the master thread.

**Comment by markballew** [comment]
Oracle is requiring everyone to login. Once that is done, you have to navigate to the java archives and download 7u80. 

I ran this script on MacOS Catalina and it works, though MacOS users will need to move the jre into place:
```
mkdir jre
mv jre1.7.0_80.jre/Contents/Home/* jre
```
Virtualmedia on Macos does not seem to work, however.

**Comment by gethvi** [comment]
The link seems to be the same and it redirects to login page if not logged in. However to download the archive it also requires accepting the Oracle License Agreement which sets a cookie. Without it the direct links don't work apparently. :-(
Thanks for pointing this out!

As for MacOS, thanks for the advice! Unfortunately I don't have any MacOS device available where I could test this and try to fix the Virtual Media.

**Comment by markballew** [comment]
I have a fix for the virtual media / keyboard issue. I'll send a PR later today, but basically https://host:443/software/avctKVMIOMac64.jar and https://host:443/software/avctVMMac64.jar need to be extracted into jre/lib or somehow included on the command line, then everything works perfectly.

**Comment by captainwasabi** [comment]
try this one, https://www.oracle.com/java/technologies/javase/javase7-archive-downloads.html

**Comment by IanWardell** [comment]
> I have a fix for the virtual media / keyboard issue. I'll send a PR later today, but basically https://host:443/software/avctKVMIOMac64.jar and https://host:443/software/avctVMMac64.jar need to be extracted into jre/lib or somehow included on the command line, then everything works perfectly.

That's a mac specific issue is it not? Are you having the issue were things like the arrow keys don't always work?

**Comment by markballew** [comment]
I sent a PR for approval. It was a one-line fix, the KVMIO jar wasn’t being extracted on MacOS.

**Comment by gethvi** [comment]
Thanks for the pull request! It was my mistake, I should have read the [master thread](https://gist.github.com/xbb/4fd651c2493ad9284dbcb827dc8886d6) more carefully, it is actually written there that MacOS has `.jnilib` extension for libraries.

**Comment by mermao1** [comment]
any fix for keyboard on Windows 64?
![image](https://user-images.githubusercontent.com/29488583/83333056-0d864c80-a275-11ea-807a-5c774ef17e22.png)

**Comment by captainwasabi** [comment]
I'm using Windows 10 Pro x64, don't have this problem.  International keyboard?

**Comment by gethvi** [comment]
Check if all the libraries were extracted successfully. After you run the script, the folder tree should like something like this (for host 10.10.10.10):

```
iDRAC6VirtualConsoleLauncher
├── iDRAC6VirtualConsoleLauncher.py
├── jre
│   └── ...
│
├── 10.10.10.10
│   └── lib
│        ├── avctKVMIO.dll
│        └── avmWinLib.dll
│
├── avctKVM.jar
├── requirements.txt
└── README.md
```

Check if you can download all of following three files from the iDRAC host. You can extract them, they are just .zip files. The first two files include `avctKVMIO.dll` and `avmWinLib.dll` libraries.
```
http://10.10.10.10/software/avctVMWin64.jar
http://10.10.10.10/software/avctKVMIOWin64.jar
http://10.10.10.10/software/avctKVM.jar
```

For me it works with Windows 10 x64, my iDRAC6 is version 2.90 and 2.92. This is my Virtual Console configuration:
```
Video Encryption Enabled:      False
Local Server Video Enabled:    True
Plug-in Type:                  Native
```

If any of this helps, please let me know.

**Comment by mermao1** [comment]
> Check if all the libraries were extracted successfully. After you run the script, the folder tree should like something like this (for host 10.10.10.10):

Yeah, I have `iDRAC6VirtualConsoleLauncher/192.168.1.10/lib/` with `avctKVMIO.dll` and `avmWinLib.dll`.


> For me it works with Windows 10 x64, my iDRAC6 is version 2.90 and 2.92. This is my Virtual Console configuration:

I just tried same configs, no luck here...
iDRAC6 2.92 in my R710 - My PC: Windows 10 x64.
I can only see the server screen (ESXi 6.7) but I can't execute commands with my keyboard.

**Comment by mermao1** [comment]
> I'm using Windows 10 Pro x64, don't have this problem. International keyboard?

yes, is a Brazilian keyboard... idk if its related but in master thread another brazillian guy had same issue https://gist.github.com/xbb/4fd651c2493ad9284dbcb827dc8886d6#gistcomment-2570699 According to nick, its a brazillian guy.

**Comment by gethvi** [comment]
If you run the script with command line `python  iDRAC6VirtualConsoleLauncher.py [options..]` it should produce quite a lot of output after the java application is launched. No hints in those logs?

**Comment by mermao1** [comment]
yeah, but according to logs, keyboard loaded complete.

log: https://gist.githubusercontent.com/mermao1/22b3d37e3f5cb8ae214450584b0affdf/raw/e29afd46d810df1c120d35453b2d73d941dde3af/log

**Comment by gethvi** [comment]
You could try download Windows image (https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/) and try to run it from VirtualBox (or whatever you prefer). It is English system by default.

You could verify that the library isn't corrupted in any way. SHA-256 for my R515 libs (idrac 2.90) and R410 libs (idrac 2.92) are the same, I assume yours should be too (just to make sure).
```
avctKVMIO.dll D0D625A4560889A3877B687CF176D0CD2AB3C2EE5C42D03722D75A8E345F4C77
avmWinLib.dll DEED3A2E0713807F5E5C173B6E2D979D9C0078E5461BD716E6A4AF18F0C1A3CE
```

But I am really running out of ideas.

**Comment by captainwasabi** [comment]
what happens if you use the windows virtual keyboard?

**Comment by mermao1** [comment]
hey guys, update: working now 😄 
I installed Dell OMSA, restarted my server with my keyboard physically attached to it, and also kept pressing the F-lock key during startup.
So far I don't know what helped, but it's working now.
Thanks for the help and support 🍻

**Comment by gethvi** [comment]
I am happy it works for you! But I really wish the fix wasn't so mysterious. :smile:
